### PR TITLE
refactor(cluster): improve logging in swim

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR contains some minimal improvements in swim
- Add separate loggers for probe, sync and gossip. This way we can adjust the log level for each separately.
- All importance state changes are logged at info level.

### CODE REVIEW

**Comment by deepthidevaki** [comment]
You can see the logs [here](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22dd-swim-logs%22%0Alabels.k8s-pod%2Fapp%3D%22camunda-platform%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fcomponent%3D%22zeebe-broker%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22dd-swim-logs%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fmanaged-by%3D%22Helm%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fname%3D%22zeebe%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fpart-of%3D%22camunda-platform%22%0AjsonPayload.context.loggerName%3D~%22io.atomix.cluster.protocol%22%0Aresource.labels.pod_name%3D%22dd-swim-logs-zeebe-0%22;timeRange=2023-06-06T09:20:03.274Z%2F2023-06-06T09:30:13.034Z;summaryFields=jsonPayload%252Fcontext%252FloggerName,resource%252Flabels%252Fpod_name:false:32:beginning;cursorTimestamp=2023-06-06T09:28:17.057886742Z?project=zeebe-io)

File: atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java
Comment (npepinpe) [inline atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java:72]: :wrench: To ensure we can always configure the loggers as a block, I would also rename this one. Either just `io.atomix.cluster.protocol.swim`, or `io.atomix.cluster.protocol.swim.protocol`.

```suggestion
  private static final Logger LOGGER = LoggerFactory.getLogger("io.atomix.cluster.protocol.swim.protocol");
```

File: atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java
Comment (npepinpe) [inline atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java:635]: :wrench: As we always probe from multiple members, I would rather combine the state into a single one at the source, i.e. `requestProbes`. Something like: " Requesting probe of Member{id=1, address=dd-swim-logs-zeebe-1.dd-swim-logs-zeebe.dd-swim-logs.svc:26502, properties={brokerInfo=EADJAAAAAwABAAAAAwAAAAMAAAADAAAAAAABCgAAAGNvbW1hbmRBcGk+AAAAZGQtc3dpbS1sb2dzLXplZWJlLTEuZGQtc3dpbS1sb2dzLXplZWJlLmRkLXN3aW0tbG9ncy5zdmM6MjY1MDEFAAMBAAAAAAIAAAAAAwAAAAEMAAIBAAAABAAAAAAAAAACAAAAAwAAAAAAAAAOAAAAOC4zLjAtU05BUFNIT1QFAAMBAAAAAQIAAAABAwAAAAE=}, version=8.3.0-SNAPSHOT, timestamp=1686042862870, state=ALIVE, incarnationNumber=1686042862876} from members Member{id=2, address=dd-swim-logs-zeebe-2.dd-swim-logs-zeebe.dd-swim-logs.svc:26502} and Member{id=3, address=dd-swim-logs-zeebe-3.dd-swim-logs-zeebe.dd-swim-logs.svc:26502}".

File: atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java
Comment (npepinpe) [inline atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java:648]: :question: I don't understand why this is info. There is no real state change here or useful information for a user here. It's useful for debugging only, no?

File: atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java:635]: :question: There's also no state change here, so I'm unsure why this is info and not debug :thinking:

**Review by npepinpe — APPROVED** [review]
:+1: Looks good! Please address my questions before merging, but I doubt we need another review

File: atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java
Comment (deepthidevaki) [reply atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java:648]: Agreed. Debug is enough here.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**
